### DISCUSSION TITLE
ZFS USEDDS Error

### DISCUSSION THREAD

**Discussion by eewiz** [openzfs/zfs#14971] [General]
Hello,

I'm new to ZFS and really need some help.
My machine:
TrueNAS Core 13.0-U5, 32GB Memory, Xeon E3-1200 4-core, 128K SSD - boot, 12 8TB spinners - pool - RAIDZ2.

There are no user files stored in the "pool" dataset.
Notice that the "pool" dataset has USEDDS=238K which I assume belongs to invisible stuff like snapshots, metadata and I don't know what else.
I assume that USEDDS=238K for the "pool" dataset is probably a normal situation.
```
root@plum[/]# zfs list -o space pool
NAME  AVAIL   USED  USEDSNAP  USEDDS  USEDREFRESERV  USEDCHILD
pool  3.37T  63.0T        0B    238K             0B      63.0T
```
There are also no user files stored in the "pool/eds" dataset either.
Notice that "pool/eds" has USEDDS=23.9T which is nowhere near correct.
```
root@plum[/]# zfs list -o space pool/eds
NAME      AVAIL   USED  USEDSNAP  USEDDS  USEDREFRESERV  USEDCHILD
pool/eds  3.37T  62.9T        0B   23.9T             0B      39.0T
```
The USEDCHILD=39.0T is correct. The "pool/eds" child datasets do add up to 39.0T.
I need to find and delete the non-visible USEDDS=23.9T data from the "pool/eds" dataset.
Here is the complete "pool/eds" dataset storage space.
```
root@plum[/]# zfs list -o space -r pool/eds
NAME               AVAIL   USED  USEDSNAP  USEDDS  USEDREFRESERV  USEDCHILD
pool/eds           3.37T  62.9T        0B   23.9T             0B      39.0T
pool/eds/backup    3.37T   881G     1.04M    881G             0B         0B
pool/eds/bt        3.37T  35.2M        0B   35.2M             0B         0B
pool/eds/common    3.37T   226G        0B    226G             0B         0B
pool/eds/drivers   3.37T   120G     36.6K    120G             0B         0B
pool/eds/fixes     3.37T  46.9G        0B   46.9G             0B         0B
pool/eds/games     3.37T  5.36T     36.6K   5.36T             0B         0B
pool/eds/keepass   3.37T   484K        0B    484K             0B         0B
pool/eds/lib       3.37T   721G     36.6K    721G             0B         0B
pool/eds/music     3.37T  1.58T     36.6K   1.58T             0B         0B
pool/eds/rch-ec    3.37T   334G     36.6K    334G             0B         0B
pool/eds/sm        3.37T   817G     36.6K    817G             0B         0B
pool/eds/software  3.37T  3.08T      951K   3.08T             0B         0B
pool/eds/urb       3.37T  1.49T        0B   1.49T             0B         0B
pool/eds/user      3.37T  66.2G      951K   66.2G             0B         0B
pool/eds/video     3.37T  24.4T     1.58G   24.4T             0B         0B
```
I have searched Google and this forum and have found no answer.
I believe that it is saying that there is 23.9TB of data stored in the "pool/eds" dataset but, ls does not show it.
```
root@plum[/]# ls -al /mnt/pool/eds
total 378
drwxrwxrwx  17 root  wheel  17 Jun 10 02:42 .
drwxrwxrwx   6 root  wheel   6 Jun 10 17:10 ..
drwxrwxrwx+ 16 root  wheel  18 Apr 30 19:37 backup
drwxrwxrwx+  2 root  wheel   3 Dec  5  2021 bt
drwxrwxrwx+ 20 root  wheel  20 Jun  5  2018 common
drwxrwxrwx+ 40 root  wheel  40 Dec  4  2019 drivers
drwxrwxrwx+ 33 root  wheel  33 Mar 26  2021 fixes
drwxrwxrwx+  9 root  wheel   9 Jan  7  2021 games
drwxrwxrwx+  2 root  wheel   3 Jun  3 01:27 keepass
drwxrwxrwx+ 29 root  wheel  32 Nov 21  2019 lib
drwxrwxrwx+ 17 root  wheel  19 Mar 30 00:13 music
drwxrwxrwx+  5 root  wheel   5 Dec  2  2021 rch-ec
drwxrwxrwx+  8 root  wheel   8 May  5  2019 sm
drwxrwxrwx+  7 root  wheel   7 Oct 23  2021 software
drwxrwxrwx+  9 root  wheel   9 Jun 10 04:36 urb
drwxrwxrwx   7 root  wheel   7 May 10 05:35 user
drwxrwxrwx+  7 root  wheel   7 Jun  4 01:02 video
root@plum[/]#
```
"ls -a" is supposed to list dot-hidden files and none show up.
"ls -d .*/" also shows no dot-hidden directories in the "pool/eds" dataset.
```
root@plum[/]# cd /mnt/pool/eds
root@plum[/mnt/pool/eds]# ls -d .*/
zsh: no matches found: .*/
root@plum[/mnt/pool/eds]#
```
Although, I do know that "/mnt/pool/eds/.zfs" exists yet, "ls -d .*/" does not show the ".zfs" folder either.
So at this point, I must assume that there is a folder in the "pool/eds" dataset that is also not visible, like ".zfs" is not visible but, consumes 23.9TB of storage space.

I don't know what I'm missing and need a little help.
Here is a description of what I did:
All of the files in /mnt/pool/eds/video had somehow been deleted and an automatic snapshot had already been created after the deletion.
I deleted that current snapshot of the empty /mnt/pool/eds/video dataset.
I cloned a one-day-old snapshot of /mnt/pool/eds/video to /mnt/pool/eds/video-clone.
I ssh'd in, then su - then I mv'd each folder from /mnt/pool/eds/video-clone/folder-name to /mnt/pool/eds/video one folder at a time.
Storage Used rose from 58% to 94% during the move. It took about 28 hours to move the full 23.9TB.
It's surprising to me that it took 28 hours to manipulate the snapshots concerned since, the move created no new actual data.
/mnt/pool/eds/video-clone was emptied and all of the data was correctly moved to /mnt/pool/eds/video.
I then went to the GUI and deleted the empty /mnt/pool/eds/video-clone.

I never made a snapshot of /mnt/pool/eds/video-clone so I expected the Storage Used to return to 58% but, it remained at 94%.
But, obviously, creating the /mnt/pool/eds/video-clone dataset must have created a system default (invisible?) snapshot.
One could create a detailed ZFS dataset structure in TrueNAS and never, ever, create a single snapshot yet the system will still work correctly.
Hence the system must create system default snapshots, even if the admin never purposefully makes any manual or automatic snapshots.
I can find absolutely no information about this, other than "deleting a dataset will also delete all snapshots for the dataset".
I assume that also means that deleting a dataset will also delete the system default snapshot including any other snapshots that were ever created.
It appears that in my case, the system default snapshot of /mnt/pool/eds/video-clone may not have been deleted when the dataset was deleted or the default snapshot was applied to the /mnt/pool/eds dataset when the /mnt/pool/eds/video-clone dataset was deleted.

I automatically make and retain snapshots for one week.
It has now been 10 days since the data move and all snapshots made before the data move have been automatically removed from the system.
I am still stuck at 94% Storage Used.
You can see above that the /mnt/pool/eds dataset shows USEDDS= 23.9T.
23.9TB is the amount of data that was cloned from the /mnt/pool/eds/video snapshot to the temporary /mnt/pool/eds/video-clone dataset.
There is no data in the /mnt/pool/eds dataset, only more datasets that are currently, correctly detailed by the USEDCHILD=39.0T value.
Actually, there never was any data in the /mnt/pool/eds dataset.
Cloning the /mnt/pool/eds/video snapshot to the /mnt/pool/eds/video-clone dataset created a new dataset in /mnt/pool/eds not USEDDS data.
The cloning to a new dataset should have increased USEDCHILD from 39TB to 63TB because the /mnt/pool/eds/video-clone dataset was simply a new dataset added to /mnt/pool/eds.
It was not 23.9TB of raw data added to /mnt/pool/eds, which would increase USEDDS to 23.9TB.
It was 23.9TB added to /mnt/pool/eds in the form of a new dataset /mnt/pool/eds/video-clone, which should have increased USEDCHILD.

I'm stuck here and don't know where to look next.
Thank You

**Comment by amotin** [comment]
You can always list all snapshots of the dataset by `zfs list -t all -r pool/eds/video`.  The moment you creates the clone you made the snapshot you created it from persistent while the clone exists.  When you copied the data you obviously increased amount of data on the pool (block cloning is not there yet).  When you deleted the clone it should now be possible to delete the snapshot, but you can make sure it really what has happened, and you don't have any older snapshots left holding the data.  According to USEDSNAP I would guess there is not much.

From the perspective of "hidden" files, one possibility for pool/eds to have space used but not visible, is that you first copied there data and then created datasets with names matching original directories.  In such case data below the children dataset mount points will still be there, but inaccessible.  To check that you should be able to unmount the datasets on top manually with umount command and check if there is anything under them.

**Comment by eewiz** [comment]
Thank you amotin.

I see below that "pool/eds/video" has a mount point showing the correct USED=24.4T.
The space used by "pool/eds/video" has increased to 24.4TB in the last 12 days since the cloning operation described above.
I don't see anything below that points to the 23.9TB of phantom data.
```
root@plum[/]# zfs list -t all -r pool/eds/video
NAME                                               USED  AVAIL     REFER  MOUNTPOINT
pool/eds/video                                    24.4T  3.34T     24.4T  /mnt/pool/eds/video
pool/eds/video@auto-2023-06-05_10-00                 0B      -     24.0T  -
pool/eds/video@auto-2023-06-06_10-00                 0B      -     24.0T  -
pool/eds/video@auto-2023-06-07_10-00                 0B      -     24.0T  -
pool/eds/video@auto-2023-06-08_10-00               457K      -     24.0T  -
pool/eds/video@auto-2023-06-09_10-00               932K      -     24.0T  -
pool/eds/video@auto-2023-06-10_10-00               457K      -     24.4T  -
pool/eds/video@auto-2023-06-11_10-00               713K      -     24.4T  -
```
eds is shorthand for "encrypted data set."
Below is the full -t all listing for the parent "eds" dataset.
USED should be 63.0T - 23.9T = 39.1T.
I don't understand what REFER=23.9T means but, that is the 23.9TB that I need to delete.

Thank you again for your invaluable help.
```
root@plum[/]# zfs list -t all -r pool/eds
NAME                                                  USED  AVAIL     REFER  MOUNTPOINT
pool/eds                                             63.0T  3.34T     23.9T  /mnt/pool/eds
pool/eds@manual-2023-06-10_20-18                        0B      -     23.9T  -
pool/eds@manual-recursive-2023-06-10_20-19              0B      -     23.9T  -
pool/eds/backup                                       881G  3.34T      881G  /mnt/pool/eds/backup
pool/eds/backup@auto-2023-06-04_16-00                   0B      -      875G  -
pool/eds/backup@auto-2023-06-05_16-00                   0B      -      875G  -
pool/eds/backup@auto-2023-06-06_16-00                   0B      -      875G  -
pool/eds/backup@auto-2023-06-07_16-00                18.3K      -      875G  -
pool/eds/backup@auto-2023-06-08_16-00                18.3K      -      875G  -
pool/eds/backup@auto-2023-06-09_16-00                   0B      -      881G  -
pool/eds/backup@auto-2023-06-10_16-00                   0B      -      881G  -
pool/eds/backup@manual-recursive-2023-06-10_20-19       0B      -      881G  -
pool/eds/bt                                          35.2M  3.34T     35.2M  /mnt/pool/eds/bt
pool/eds/bt@manual-recursive-2023-06-10_20-19           0B      -     35.2M  -
pool/eds/common                                       226G  3.34T      226G  /mnt/pool/eds/common
pool/eds/common@auto-2023-06-04_15-30                   0B      -      226G  -
pool/eds/common@auto-2023-06-05_15-30                   0B      -      226G  -
pool/eds/common@auto-2023-06-06_15-30                   0B      -      226G  -
pool/eds/common@auto-2023-06-07_15-30                   0B      -      226G  -
pool/eds/common@auto-2023-06-08_15-30                   0B      -      226G  -
pool/eds/common@auto-2023-06-09_15-30                   0B      -      226G  -
pool/eds/common@auto-2023-06-10_15-30                   0B      -      226G  -
pool/eds/common@manual-recursive-2023-06-10_20-19       0B      -      226G  -
pool/eds/drivers                                      120G  3.34T      120G  /mnt/pool/eds/drivers
pool/eds/drivers@auto-2023-06-04_15-00                  0B      -      120G  -
pool/eds/drivers@auto-2023-06-05_15-00                  0B      -      120G  -
pool/eds/drivers@auto-2023-06-06_15-00                  0B      -      120G  -
pool/eds/drivers@auto-2023-06-07_15-00               18.3K      -      120G  -
pool/eds/drivers@auto-2023-06-08_15-00                  0B      -      120G  -
pool/eds/drivers@auto-2023-06-09_15-00                  0B      -      120G  -
pool/eds/drivers@auto-2023-06-10_15-00                  0B      -      120G  -
pool/eds/drivers@manual-recursive-2023-06-10_20-19      0B      -      120G  -
pool/eds/fixes                                       46.9G  3.34T     46.9G  /mnt/pool/eds/fixes
pool/eds/fixes@auto-2023-06-04_14-30                    0B      -     46.9G  -
pool/eds/fixes@auto-2023-06-05_14-30                    0B      -     46.9G  -
pool/eds/fixes@auto-2023-06-06_14-30                    0B      -     46.9G  -
pool/eds/fixes@auto-2023-06-07_14-30                    0B      -     46.9G  -
pool/eds/fixes@auto-2023-06-08_14-30                    0B      -     46.9G  -
pool/eds/fixes@auto-2023-06-09_14-30                    0B      -     46.9G  -
pool/eds/fixes@auto-2023-06-10_14-30                    0B      -     46.9G  -
pool/eds/fixes@manual-recursive-2023-06-10_20-19        0B      -     46.9G  -
pool/eds/games                                       5.36T  3.34T     5.36T  /mnt/pool/eds/games
pool/eds/games@auto-2023-06-04_14-00                    0B      -     5.36T  -
pool/eds/games@auto-2023-06-05_14-00                    0B      -     5.36T  -
pool/eds/games@auto-2023-06-06_14-00                    0B      -     5.36T  -
pool/eds/games@auto-2023-06-07_14-00                 18.3K      -     5.36T  -
pool/eds/games@auto-2023-06-08_14-00                    0B      -     5.36T  -
pool/eds/games@auto-2023-06-09_14-00                    0B      -     5.36T  -
pool/eds/games@auto-2023-06-10_14-00                    0B      -     5.36T  -
pool/eds/games@manual-recursive-2023-06-10_20-19        0B      -     5.36T  -
pool/eds/games@auto-2023-06-11_14-00                    0B      -     5.36T  -
pool/eds/keepass                                      484K  3.34T      484K  /mnt/pool/eds/keepass
pool/eds/keepass@auto-2023-06-05_13-30                  0B      -      484K  -
pool/eds/keepass@auto-2023-06-06_13-30                  0B      -      484K  -
pool/eds/keepass@auto-2023-06-07_13-30                  0B      -      484K  -
pool/eds/keepass@auto-2023-06-08_13-30                  0B      -      484K  -
pool/eds/keepass@auto-2023-06-09_13-30                  0B      -      484K  -
pool/eds/keepass@auto-2023-06-10_13-30                  0B      -      484K  -
pool/eds/keepass@manual-recursive-2023-06-10_20-19      0B      -      484K  -
pool/eds/keepass@auto-2023-06-11_13-30                  0B      -      484K  -
pool/eds/lib                                          721G  3.34T      721G  /mnt/pool/eds/lib
pool/eds/lib@auto-2023-06-05_13-00                      0B      -      721G  -
pool/eds/lib@auto-2023-06-06_13-00                      0B      -      721G  -
pool/eds/lib@auto-2023-06-07_13-00                      0B      -      721G  -
pool/eds/lib@auto-2023-06-08_13-00                      0B      -      721G  -
pool/eds/lib@auto-2023-06-09_13-00                      0B      -      721G  -
pool/eds/lib@auto-2023-06-10_13-00                      0B      -      721G  -
pool/eds/lib@manual-recursive-2023-06-10_20-19          0B      -      721G  -
pool/eds/lib@auto-2023-06-11_13-00                      0B      -      721G  -
pool/eds/music                                       1.58T  3.34T     1.58T  /mnt/pool/eds/music
pool/eds/music@auto-2023-06-05_12-30                    0B      -     1.58T  -
pool/eds/music@auto-2023-06-06_12-30                    0B      -     1.58T  -
pool/eds/music@auto-2023-06-07_12-30                    0B      -     1.58T  -
pool/eds/music@auto-2023-06-08_12-30                    0B      -     1.58T  -
pool/eds/music@auto-2023-06-09_12-30                    0B      -     1.58T  -
pool/eds/music@auto-2023-06-10_12-30                    0B      -     1.58T  -
pool/eds/music@manual-recursive-2023-06-10_20-19        0B      -     1.58T  -
pool/eds/music@auto-2023-06-11_12-30                    0B      -     1.58T  -
pool/eds/rch-ec                                       334G  3.34T      334G  /mnt/pool/eds/rch-ec
pool/eds/rch-ec@auto-2023-06-05_12-00                   0B      -      334G  -
pool/eds/rch-ec@auto-2023-06-06_12-00                   0B      -      334G  -
pool/eds/rch-ec@auto-2023-06-07_12-00                   0B      -      334G  -
pool/eds/rch-ec@auto-2023-06-08_12-00                   0B      -      334G  -
pool/eds/rch-ec@auto-2023-06-09_12-00                   0B      -      334G  -
pool/eds/rch-ec@auto-2023-06-10_12-00                   0B      -      334G  -
pool/eds/rch-ec@manual-recursive-2023-06-10_20-19       0B      -      334G  -
pool/eds/rch-ec@auto-2023-06-11_12-00                   0B      -      334G  -
pool/eds/sm                                           817G  3.34T      817G  /mnt/pool/eds/sm
pool/eds/sm@auto-2023-06-05_11-30                       0B      -      817G  -
pool/eds/sm@auto-2023-06-06_11-30                       0B      -      817G  -
pool/eds/sm@auto-2023-06-07_11-30                       0B      -      817G  -
pool/eds/sm@auto-2023-06-08_11-30                       0B      -      817G  -
pool/eds/sm@auto-2023-06-09_11-30                       0B      -      817G  -
pool/eds/sm@auto-2023-06-10_11-30                       0B      -      817G  -
pool/eds/sm@manual-recursive-2023-06-10_20-19           0B      -      817G  -
pool/eds/sm@auto-2023-06-11_11-30                       0B      -      817G  -
pool/eds/software                                    3.08T  3.34T     3.08T  /mnt/pool/eds/software
pool/eds/software@auto-2023-06-05_11-00                 0B      -     3.06T  -
pool/eds/software@auto-2023-06-06_11-00                 0B      -     3.06T  -
pool/eds/software@auto-2023-06-07_11-00              18.3K      -     3.06T  -
pool/eds/software@auto-2023-06-08_11-00              18.3K      -     3.08T  -
pool/eds/software@auto-2023-06-09_11-00                 0B      -     3.08T  -
pool/eds/software@auto-2023-06-10_11-00                 0B      -     3.08T  -
pool/eds/software@manual-recursive-2023-06-10_20-19     0B      -     3.08T  -
pool/eds/software@auto-2023-06-11_11-00                 0B      -     3.08T  -
pool/eds/urb                                         1.51T  3.34T     1.51T  /mnt/pool/eds/urb
pool/eds/urb@manual-recursive-2023-06-10_20-19       1.15G      -     1.49T  -
pool/eds/user                                        66.2G  3.34T     66.2G  /mnt/pool/eds/user
pool/eds/user@auto-2023-06-05_10-30                   585K      -     66.2G  -
pool/eds/user@auto-2023-06-06_10-30                   347K      -     66.2G  -
pool/eds/user@auto-2023-06-07_10-30                     0B      -     66.2G  -
pool/eds/user@auto-2023-06-08_10-30                     0B      -     66.2G  -
pool/eds/user@auto-2023-06-09_10-30                     0B      -     66.2G  -
pool/eds/user@auto-2023-06-10_10-30                     0B      -     66.2G  -
pool/eds/user@manual-recursive-2023-06-10_20-19         0B      -     66.2G  -
pool/eds/user@auto-2023-06-11_10-30                     0B      -     66.2G  -
pool/eds/video                                       24.4T  3.34T     24.4T  /mnt/pool/eds/video
pool/eds/video@auto-2023-06-05_10-00                    0B      -     24.0T  -
pool/eds/video@auto-2023-06-06_10-00                    0B      -     24.0T  -
pool/eds/video@auto-2023-06-07_10-00                    0B      -     24.0T  -
pool/eds/video@auto-2023-06-08_10-00                  457K      -     24.0T  -
pool/eds/video@auto-2023-06-09_10-00                  932K      -     24.0T  -
pool/eds/video@auto-2023-06-10_10-00                  457K      -     24.4T  -
pool/eds/video@manual-recursive-2023-06-10_20-19      494K      -     24.4T  -
pool/eds/video@auto-2023-06-11_10-00                  713K      -     24.4T  -
```

**Reply by rincebrain** [reply]
`REFER`, briefly, means "how much data this copy of the filesystem/volume whatever has references to".

So if it says 23.9T `REFER` on the live copy of the filesystem, it means it thinks there are 23.9T of actual data on the filesystem and not its children, or snapshots, or anywhere else.

You could, one imagines, use `zdb` to poke at where it thinks that space is, and the filenames involved, with something like `zdb -dddddd pool/eds`, but that would be very, very verbose indeed...

I would strongly bet, as amotin suggested above, that you have data in folders on `pool/eds` that have things mounted on them, such that you can't see them, but they're there, and would try unmounting every mountpoint under `eds` and then running `du -ax` again.

**Comment by eewiz** [comment]
Hello All

I unmounted all of the datasets below /mnt/pool/eds to find the following:
```
root@plum[/mnt/pool/eds]# ls -l
total 285
drwxr-xr-x  2 root  wheel  2 Nov 14  2021 backup
drwxr-xr-x  2 root  wheel  2 Nov 13  2021 bt
drwxr-xr-x  2 root  wheel  2 Nov 15  2021 common
drwxr-xr-x  2 root  wheel  2 Nov 14  2021 drivers
drwxr-xr-x  2 root  wheel  2 Nov 14  2021 fixes
drwxr-xr-x  2 root  wheel  2 Nov 17  2021 games
drwxr-xr-x  2 root  wheel  2 Dec  2  2021 keepass
drwxr-xr-x  2 root  wheel  2 Nov 19  2021 lib
drwxr-xr-x  2 root  wheel  2 Nov 21  2021 music
drwxr-xr-x  2 root  wheel  2 Nov 21  2021 rch-ec
drwxr-xr-x  2 root  wheel  2 Nov 23  2021 sm
drwxr-xr-x  2 root  wheel  2 Nov 20  2021 software
drwxr-xr-x  2 root  wheel  2 Nov 13  2021 urb
drwxr-xr-x  2 root  wheel  2 Nov 14  2021 user
drwxrwxrwx+ 7 root  wheel  7 Jun  1 07:08 video
```
The above are all folders, not datasets.
I originally thought that the unmounts did not work until I noticed all of the permissions had changed, except for the "video" folder.
Ultimately, I had to share the "eds" dataset to view it in Windows before I could determine that what is seen above are folders, not datasets.
I see no way to tell the difference from an "ls -l" output.

On day-one, two years ago, I must have created folders in the /mnt/pool/eds dataset only to later discover that folders cannot be replicated.
Since folders residing in a dataset are totally invisible in the TrueNAS GUI there was no visual indication, nor warning thrown, that datasets were being created in the TrueNAS GUI with the same name as existing folders.
All of the folders were empty except for the /mnt/pool/eds/video folder which held the 23.9TB in question.
I renamed the video folder to "video_dir" and rebooted TrueNAS.
After verifying that all the datasets were correct, I started an "rm -R video_dir"

I understand that the [clone-snap, move-data, delete-empty-clone] method I employed did not work but now, I wonder why.

Should I have not done [clone-snap, move-data, delete-empty-clone] or was the problem caused by the underlying, same-named folder?
I don't understand why the data was actually moved from the "video-clone" dataset to the "video" folder and not the "video" dataset.
I expected that ZFS would have just moved the file metadata from the "video-clone" dataset's metadata table (no snapshot was ever created) to the "video" dataset's snapshot like a move in Windows simply changes the file allocation table, or whatever it's called these days in Windows, leaving the actual file data untouched.
I expected the move to take no time at all because, I expected that no new data would be created.

I am curious, if the underlying "video" folder did not exist, would the [clone-snap, move-data, delete-empty-clone] method have worked?

Originally, I simply deleted the last snapshot, which was auto-created after all of the files in the "video" dataset were deleted, then rebooted.
This did not return the "video" dataset to normal, it was still empty.
Then I tried to ROLLBACK the now last, good snapshot, taken the day before the files were deleted and nothing happened.
The "video" dataset remained empty.
Then I did the [clone-snap, move-data, delete-empty-clone] maneuver to ultimately discover here today that the data was moved to the "video" folder instead of the "video" dataset.

I can now report that the rm was successful.
After a reboot, the server is now back to 59% Used Space.

Thank You
